staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/build_controller.go

```diff
@@ -1776,16 +1828,21 @@ func (bc *BuildController) handleControllerConfig() []error {
 	if err != nil {
 		configErrs = append(configErrs, err)
 	}
+	err = bc.readClusterProxyConfig()
+	if err != nil {
+		configErrs = append(configErrs, err)
+	}
```

Comment:
`readClusterProxyConfig` should be run before `readClusterBuildControllerConfig`. We want the precedence to be as follows:

1. Set the build & git proxy from the global cluster config
2. If cluster build config has the `DefaultProxy` set, override the build & git proxy
3. If the cluster build config has the `GitProxy` set, override the git proxy.

Note that the build controller config API uses pointers and `omitempty`, so we can distinguish between unset (`nil`) and empty/don't use a proxy.

Reply:
Ah good catch I had meant to maintain that path when moving to the common cfg handler

will update